framework/assets/yii.gridView.js

```diff
@@ -85,25 +85,40 @@
             var $grid = $(this), event;
             var settings = gridData[$grid.attr('id')].settings;
             var data = {};
-            $.each($(settings.filterSelector).serializeArray(), function () {
-                data[this.name] = this.value;
-            });
+            var dataForm = [];
 
+            // get the query params from the request url
             $.each(yii.getQueryParams(settings.filterUrl), function (name, value) {
-                if (data[name] === undefined) {
-                    data[name] = value;
+                data[name.replace(/\[[0-9]+\]/i, '[]')] = value;
```

Comment:
It could be a good solution but I think it's better to have that hacky code in the JS than implement a custom core function from the PHP. We could remove them in the response of the http_build_query() so it will be easier to maintain.